File: 001-tdd-wprowadzenie/Java/src/main/java/lukaszstaniszewski/StrongCalc.java
package lukaszstaniszewski;

/**
 * The type Strong calc.
 */
public class StrongCalc {

    /**
     * Calc int.
     *
     * @param strong the strong
     * @return the int
     * @throws Exception the exception
     */
    public int calc(int strong) throws Exception{
        if (strong <= 0) {
            throw new Exception();
        }

        int strongResult = 1;

        for (int i = 1; i <= strong; ++i) {
            strongResult *= i;
        }

        return strongResult;
    }

}

File: 001-tdd-wprowadzenie/PHP/src/StrongCalc.php
<?php
declare(strict_types=1);

namespace App;

use \Exception;

/**
 * Class StrongCalc
 * @package App
 */
class StrongCalc
{
    /**
     * @param int $strong
     * @return int
     * @throws Exception
     */
    public function calc(int $strong): int
    {
        if ($strong <= 0) {
            throw new Exception("Strong can't be negative!");
        }

        $strongResult = 1;

        for ($i = 1; $i <= $strong; ++$i) {
            $strongResult *= $i;
        }

        return $strongResult;
    }
}
